src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/notification/NotificationEventBatchSubscriber.kt

```diff
@@ -0,0 +1,87 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.message.impl.rabbitmq.notification
+
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.schema.message.FilterFunction
+import com.exactpro.th2.common.schema.message.MessageListener
+import com.exactpro.th2.common.schema.message.MessageSubscriber
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.configuration.SubscribeTarget
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.ConnectionManager
+import com.rabbitmq.client.Delivery
+import mu.KotlinLogging
+import java.util.concurrent.CopyOnWriteArrayList
+
+class NotificationEventBatchSubscriber(
+    private val connectionManager: ConnectionManager,
+    private val queue: String
+) : MessageSubscriber<EventBatch> {
+    private val listeners = CopyOnWriteArrayList<MessageListener<EventBatch>>()
+
+    @Deprecated(
+        "Method is deprecated, please use constructor",
+        ReplaceWith("NotificationEventBatchSubscriber()")
+    )
+    override fun init(connectionManager: ConnectionManager, exchangeName: String, subscribeTargets: SubscribeTarget) {
+        throw UnsupportedOperationException("Method is deprecated, please use constructor")
+    }
+
+    @Deprecated(
+        "Method is deprecated, please use constructor",
+        ReplaceWith("NotificationEventBatchSubscriber()")
+    )
+    override fun init(
+        connectionManager: ConnectionManager,
+        subscribeTarget: SubscribeTarget,
+        filterFunc: FilterFunction
+    ) {
+        throw UnsupportedOperationException("Method is deprecated, please use constructor")
+    }
+
+    override fun start() {
+        connectionManager.basicConsume(
+            queue,
+            { consumerTag: String, delivery: Delivery ->
+                for (listener in listeners) {
+                    try {
+                        listener.handler(consumerTag, EventBatch.parseFrom(delivery.body))
+                    } catch (listenerExc: Exception) {
+                        LOGGER.warn(
+                            "Message listener from class '{}' threw exception",
+                            listener.javaClass,
+                            listenerExc
+                        )
+                    }
+                }
+            },
+            { LOGGER.warn("Consuming cancelled for: '{}'", it) }
+        )
+    }
+
+    override fun addListener(messageListener: MessageListener<EventBatch>) {
+        listeners.add(messageListener)
+    }
+
+    override fun close() {
+        listeners.forEach(MessageListener<EventBatch>::onClose)
+        listeners.clear()
```

Comment:
Please call the Channel.basicCancel(consumerTag) method to unsubscribe

Reply:
done